modules/ROOT/pages/sample-data-java-sdk.adoc

```diff
@@ -0,0 +1,472 @@
+
+   @Parameter
+   private String objectType;
+
+   @Parameter
+   @Optional
+   private String status;
+
+   @Connection
+   private GitHubConnection connection;
+
+   @Config
+   private GitHubConfig config;
+
+   @Override
+   public Result<InputStream, ObjectAttributes> getSample() throws SampleDataException {
+       String query = "objectType=" + objectType;
+       if (status != null) {
+           query += "; status=" + status;
+       }
+       query += ";organization=" + config.getOrganization();
+
+       List<GitHubObject> jsonDocs = connection.query(query);
+
+       if (jsonDocs.isEmpty()) {
+           throw new SampleDataException("No data available", NO_DATA_AVAILABLE);
+       }
+
+       GitHubObject sample = jsonDocs.get(0);
+       String body = sample.getBody();
+
+       return Result.<InputStream, ObjectAttributes>builder()
+               .output(new ByteArrayInputStream(body.getBytes()))
+               .mediaType(MediaType.APPLICATION_JSON)
+               .length(body.length())
+               .attributes(sample.getMetadata())
+               .attributesMediaType(MediaType.APPLICATION_JAVA)
+               .build();
+   }
+
+   @Override
+   public String getId() {
+       return "objectResolver";
+   }
+}
+----
+
+Both components have the same type signature as the sample data provider. They
+define the same parameters that the resolver needs, but they both have
+different additional parameters that the resolver does not require. The resolver
+accesses the parameters, connection, and configuration the same way a
+value provider does.
+
+If no data is available, the resolver throws a *SampleDataException* with a
+defined failure code.
+
+== OAuth Token Refresh
+
+If your operation or source uses an OAuth connection, trigger a token
+refresh in the provider by throwing an *AccessTokenExpiredException* inside
+the getSample() method, as done for
+https://docs.mulesoft.com/mule-sdk/1.1/oauth-token-expiration[operations or sources].
+
+The following example illustrates a token refresh:
+
+[source,java,linenums]
+----
+public class RefreshedOAuthSampleDataProvider implements SampleDataProvider<String, String> {
+
+ public static final String SAMPLE_PAYLOAD_VALUE = "Sample payload!";
+ public static final String SAMPLE_ATTRIBUTES_VALUE = "Sample Attributes!";
+
+ @Connection
+ private TestOAuthConnection testOAuthConnection;
+
+ @Override
+ public String getId() {
+   return "OAuth sample data";
+ }
+
+ @Override
+ public Result getSample() throws SampleDataException {
+   if (!testOAuthConnection.getState().getAccessToken().contains("refresh")) {
+     throw new AccessTokenExpiredException();
+   }
+   return builder()
+       .output(SAMPLE_PAYLOAD_VALUE)
+       .attributes(SAMPLE_ATTRIBUTES_VALUE)
+       .build();
+ }
+}
+----
+
+== Aliases and Acting Parameters
+
+Parameter resolution works in the same way as value providers. Aliases are
```

Comment:
"resolution" and "providers" are not equal objects, so how can they work in the same way?
Parameter resolution is a process; value providers are entities.

Reply:
fixed